projects/flow.groups-2D-random-walk/src/project.py

```diff
@@ -0,0 +1,211 @@
+"""Define the operations for the 2D Gaussian Random Walk project."""
+
+import flow
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+# Helper functions
+def get_replica(jobs, replica):
+    """Get a certain replica from a group of jobs."""
+    try:
+        job = next(filter(lambda job: job.sp.replica == replica, jobs))
+    except StopIteration:
+        raise RuntimeError(f"No replica {replica} found.")
+    return job
+
+
+def generate_stores(jobs, store_name):
+    """Yield a data store for each job in jobs."""
+    for job in jobs:
+        with job.data:
+            yield job.data[store_name]
+
+
+class RandomWalkProject(flow.FlowProject):
+    """Create a workflow for simulating 2D Gaussian random walks."""
+
+    pass
+
+
+@RandomWalkProject.label
+def simulated(job):
+    """Return whether the job simulated."""
+    return "positions" in job.data
+
+
+def all_simulated(*jobs):
+    """Return whether all jobs simulated."""
+    return all(simulated(job) for job in jobs)
+
+
+# Operations on individual jobs
+
+# Group for operations on single jobs that are reachable from simulate
+base = RandomWalkProject.make_group("base")
+
+
+@base
+@RandomWalkProject.post(simulated)
+@RandomWalkProject.operation
+def simulate(job):
+    """Simulate a 2D random walk."""
+    # use job id as random seed
+    n_steps = job.doc.run_steps
+    generator = np.random.default_rng(int(job.id, 16))
```

Comment:
I don't think this is very reproducible. The presence of unused parameters in the state point, for instance, should not affect the seed. As a matter of best practice, I'd prefer to show users that they should explicitly generate and save RNG seeds as state point parameters.

Reply:
That depends on what kind of reproducibility you are looking for; if you want "trajectory" level reproducibility then no, but if you are just looking for getting the same statistics then this is perfectly reproducible. HOOMD for instance doesn't promise the same trajectory output even when using the exact same random seed on the same hardware.